test/call.c

```diff
@@ -884,6 +912,102 @@ int test_call_video(void)
 }
 
 
+int test_call_change_videodir(void)
+{
+	struct fixture fix, *f = &fix;
+	struct vidisp *vidisp = NULL;
+	enum sdp_dir a_video_ldir, a_video_rdir, b_video_ldir, b_video_rdir;
+	int err = 0;
+
+	conf_config()->video.fps = 100;
+	conf_config()->video.enc_fmt = VID_FMT_YUV420P;
+
+	fixture_init(f);
+
+	/* to enable video, we need one vidsrc and vidcodec */
+	mock_vidcodec_register();
+
+	err = mock_vidisp_register(&vidisp, mock_vidisp_handler, f);
+	TEST_ERR(err);
+
+	err = module_load(".", "fakevideo");
+	TEST_ERR(err);
+
+	f->behaviour = BEHAVIOUR_ANSWER;
+	f->estab_action = ACTION_RECANCEL;
+	f->accept_session_updates = true;
+
+	/* Make a call from A to B */
+	err = ua_connect(f->a.ua, 0, NULL, f->buri, VIDMODE_ON);
+	TEST_ERR(err);
+
+	/* run main-loop with timeout, wait for events */
+	err = re_main_timeout(10000);
+	TEST_ERR(err);
+	TEST_ERR(fix.err);
+
+	/* verify that video was enabled for this call */
+	ASSERT_EQ(1, fix.a.n_established);
+	ASSERT_EQ(1, fix.b.n_established);
+
+	ASSERT_TRUE(call_has_video(ua_call(f->a.ua)));
+	ASSERT_TRUE(call_has_video(ua_call(f->b.ua)));
+
+	/* Set video inactive */
```

Comment:
Set "state" here to "inactive requested".

Reply:
See other comment regarding state.